Toggl.Daneel/ViewControllers/LoginViewController.cs

```diff
@@ -41,226 +32,233 @@ public override void ViewDidLoad()
-        }
+            ViewModel.PasswordRelay
+                .Subscribe(PasswordTextField.Rx().TextObserver())
+                .DisposedBy(DisposeBag);
 
-        public override void ViewWillDisappear(bool animated)
-        {
-            base.ViewWillDisappear(animated);
-            NavigationController?.SetNavigationBarHidden(false, true);
-        }
+            ViewModel.LoginWithEmail.Errors
+                .Select(e => e.Message)
+                .Subscribe(LoginWithEmailErrorLabel.Rx().Text())
+                .DisposedBy(DisposeBag);
 
-        public override void ViewDidLayoutSubviews()
-        {
-            base.ViewDidLayoutSubviews();
+            ViewModel.IsLoggingIn
+                .Subscribe(ActivityIndicator.Rx().IsVisibleWithFade())
+                .DisposedBy(DisposeBag);
 
-            if (View.Frame.Height > iPhoneSeScreenHeight && !keyboardIsOpen)
-                TopConstraint.Constant = topConstraintForBiggerScreens;
+            ViewModel.IsLoggingIn
+                .Subscribe(UIApplication.SharedApplication.Rx().NetworkActivityIndicatorVisible())
+                .DisposedBy(DisposeBag);
 
-            SignupCard.SetupBottomCard();
-            GoogleLoginButton.SetupGoogleButton();
-        }
+            ViewModel.IsLoggingIn.Select(loginButtonTitle)
+                .Subscribe(LoginButton.Rx().AnimatedTitle())
+                .DisposedBy(DisposeBag);
 
-        public override void ViewDidAppear(bool animated)
-        {
-            base.ViewDidAppear(animated);
-            ActivityIndicator.Alpha = 0;
-            ActivityIndicator.StartSpinning();
-            PasswordTextField.ResignFirstResponder();
-        }
+            ViewModel.IsInSecondScreen
+                .Invert()
+                .Do(isGoingBack =>
+                {
+                    if (isGoingBack)
+                    {
+                        LoginWithEmailTextField.BecomeFirstResponder();
+                    }
+                })
+                .Subscribe(FirstScreenWrapperView.Rx().AnimatedIsVisible());
 
-        private void KeyboardWillShow(object sender, UIKeyboardEventArgs e)
-        {
-            keyboardIsOpen = true;
-            if (View.Frame.Height > iPhoneSeScreenHeight)
-            {
-                TopConstraint.Constant = topConstraintForBiggerScreensWithKeyboard;
-            }
-            else
-            {
-                EmailFieldTopConstraint.Constant = emailTopConstraintWithKeyboard;
-            }
-            UIView.Animate(Animation.Timings.EnterTiming, () => View.LayoutIfNeeded());
-        }
+            ViewModel.IsInSecondScreen
+                .Do(isSecondScreen =>
+                {
+                    if (isSecondScreen)
+                    {
+                        PasswordTextField.BecomeFirstResponder();
+                    }
+                })
+                .Subscribe(SecondScreenWrapperView.Rx().AnimatedIsVisible());
```

Comment:
Missing `DisposedBy` here.
```suggestion
                .Subscribe(SecondScreenWrapperView.Rx().AnimatedIsVisible())
                .DisposedBy(DisposeBag);
```